### PR TITLE
tracing: Add Jaeger tracing example

### PR DESCRIPTION
**Pull request by objectiser**

### CODE REVIEW

**Comment by mattklein123** [comment]
@objectiser wow cool! Is there any other Jaeger person that could help review this? We can give it a quick review for grammar, config sanity, etc. but don't really have any context to do a tracing specific review.

**Comment by moderation** [comment]
I haven't tested the sandbox in this PR but I'm successfully sending traces to Jaeger after this PR merged that allows Jaeger to ingest JSON - https://github.com/uber/jaeger/commit/84f1a689cc8a0533da88c653c6c1b013a2199c0e

**Comment by mattklein123** [comment]
@moderation if you feel like testing this sandbox and giving it a +1 that would be cool. No pressure. :)

**Comment by moderation** [comment]
@mattklein123 sandboxes are designed for people unfortunate enough to be running MacOS :-) Will see if I can take a look a bit later.

File: examples/jaeger-tracing/front-envoy-jaeger.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            "codec_type": "auto",
+            "stat_prefix": "ingress_http",
+            "route_config": {
+              "virtual_hosts": [
+                {
+                  "name": "backend",
+                  "domains": ["*"],
+                  "routes": [
+                    {
+                      "timeout_ms": 0,
```
Comment (RomanDzhabarov) [inline examples/jaeger-tracing/front-envoy-jaeger.json]: i'd probably remove timeout_ms setting and also for simplicity you can say "prefix": "/".

File: examples/jaeger-tracing/service1-envoy-jaeger.json [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            "codec_type": "auto",
+            "stat_prefix": "ingress_http",
+            "route_config": {
+              "virtual_hosts": [
+                {
+                  "name": "service1",
+                  "domains": ["*"],
+                  "routes": [
+                    {
+                      "timeout_ms": 0,
```
Comment (RomanDzhabarov) [inline examples/jaeger-tracing/service1-envoy-jaeger.json]: nit: i'd remove this.

File: examples/jaeger-tracing/front-envoy-jaeger.json (hunk 1 above)
Comment (objectiser) [inline examples/jaeger-tracing/front-envoy-jaeger.json]: Was wondering why such a specific path had been defined - I've also make the same changes to the zipkin example and tested both.

**Comment by jpkrohling** [comment]
Using Jaeger as a drop-in replacement for Zipkin looks correct to me as a first step. I tested the instructions and got the trace reported correctly when using the latest `jaegertracing/all-in-one` image:

image

File: examples/jaeger-tracing/front-envoy-jaeger.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{
+  "listeners": [
+    {
+      "address": "tcp://0.0.0.0:80",
+      "filters": [
+        {
+          "type": "read",
+          "name": "http_connection_manager",
+          "config": {
+            "generate_request_id": true,
```
Comment (mattklein123) [inline examples/jaeger-tracing/front-envoy-jaeger.json]: nit: this is the default so can be removed.

**Pull request merged**